language/move-prover/boogie-backend/src/boogie_wrapper.rs

```diff
@@ -1682,7 +1682,7 @@ impl<'s> ModelParser<'s> {
 
     fn parse_key(&mut self) -> Result<ModelValue, ModelParseError> {
         let mut comps = vec![];
-        while !self.looking_at("->") {
+        while !self.looking_at("->") && self.at < self.input.len() {
```

Comment:
I am running Ubuntu 20.04.4 LTS and I realize that my prover tests pass without this line of change...

Reply:
So I still don't quite understand what this line is doing...

Reply:
Sud0u53r at our Discord channel points out that this is a bug which causes Prover fall into an infinite loop when `boogie_wrapper` tries to parse an incomplete map expression with no value part (like `var ->`). This is a clear bug that needs to be fixed.

Reply:
> I am running Ubuntu 20.04.4 LTS and I realize that my prover tests pass without this line of change...
If you upgrade Boogie version to 2.14.0, Boogie will start producing incomplete expressions like (`$abort_code ->`). If you run Prover again without this fix, Prover will not terminate due to the infinite loop bug. If you run Prover with this fix, Prover will terminate, but complain about such incomplete expressions.

I am trying to be confirmed whether such incomplete expressions are the expected behavior of Boogie or a bug. In the meantime, we can safely upgrade and use the Boogie version of 2.13.4.

Reply:
I see. Fair enough